src/Model/NotificationModel.php

```diff
@@ -13,16 +13,10 @@
 
 class NotificationModel implements NotificationInterface
 {
-    private $id;
-    private $message;
-    private $type;
-    private $url;
+    private ?string $url = null;
 
-    public function __construct(string $id, string $message, string $type = Constants::TYPE_INFO)
+    public function __construct(private string $id, private string $message, private string $type = Constants::TYPE_INFO)
```

Comment:
I'm not a big fan of putting class members INSIDE the constructor.
We have encountered a lot of confusing behavior logic when using it that way (in our company).
Even more in a single line! It will be a pain when resolving conflicts on merge! (trust me 😭)

No complaint when using autowire though, just on models like this one.

I'll go:
```php
public function __construct(
    private string $id, 
    private string $message, 
    private string $type = Constants::TYPE_INFO
) {
}
```

Or back to OLD way
```php
private string $id, 
private string $message, 
private string $type = Constants::TYPE_INFO
private ?string $url = null;

public function __construct(
    string $id, 
    string $message, 
    string $type = Constants::TYPE_INFO
) {
        $this->id = $id;
        $this->message = $message;
        $this->type = $type;
}
```

Reply:
If you are using `rector` to do this PR, we have added the `Rector\Php80\Rector\Class_\ClassPropertyAssignToConstructorPromotionRector` to the `skip` config. And do this upgrade only where needed.